pre_commit_hooks/format_cspell_json.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 TIER IV, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import json
+from pathlib import Path
+from typing import Dict, Optional, Sequence
+
+
+def format_json_recursive(j: Dict) -> Dict:
+    if isinstance(j, dict):
+        j = {k: format_json_recursive(v) for k, v in j.items()}
+    elif isinstance(j, list):
+        if any([not isinstance(v, str) for v in j]):
+            j = [format_json_recursive(v) for v in j]
+        else:
+            j = list(set(j))
+            j = sorted(j, key=str.lower)
+    return j
+
+
+def format_cspell_json(cspell_json: Dict) -> Dict:
+    # Sort inside each key
+    formatted_json = format_json_recursive(cspell_json)
+
+    # Sort top-level keys
+    formatted_json = {
+        "version": formatted_json["version"],
+        "language": formatted_json["language"],
```

Comment:
`languageSettings` is intentionally removed because it seemed unnecessary.
It just has `allowCompoundWords: false`, which is also defined in the top-level.

Reply:
Reference: https://github.com/streetsidesoftware/cspell/tree/main/packages/cspell#LanguageSettings